File: sort/sort.py
class Sorting:
    def __init__(self):
        self.array = []

    def enterArray(self):
        userInput = input("Podaj tablice 6 znakow po przecinku: ")
        self.array = [int(x.strip()) for x in userInput.split(",") if x.strip()]
        self.n = len(self.array)

    def bubbleSort(self):
        for i in range(self.n - 1):
            for j in range(self.n-1-i):
                if (self.array[j] > self.array[j+1]):
                    self.array[j], self.array[j+1] = self.array[j+1], self.array[j]
        return self.array

    def selectionSort(self):
        for i in range(self.n):
            min_index = i
            for j in range(i + 1, self.n):
                if (self.array[j] < self.array[min_index]):
                    min_index = j
            self.array[i], self.array[min_index] = self.array[min_index], self.array[i]
        return self.array
    
    def insertionSort(self):
        for i in range(1, self.n):
            key = self.array[i]
            j = i - 1
            while j >= 0 and key < self.array[j]:
                self.array[j + 1] = self.array[j]
                j -= 1
            self.array[j+1] = key
        return self.array

    def mergeSort(self, arr=None):
        if arr is None:
            arr = self.array

        if len(arr) > 1:
            mid = len(arr) // 2
            left = arr[0:mid]
            right = arr[mid:len(arr)]

            self.mergeSort(left)
            self.mergeSort(right)
            
            i = j = k = 0

            while i < len(left) and j < len(right):
                if left[i] < right[j]:
                    arr[k] = left[i]
                    i += 1
                else:
                    arr[k] = right[j]
                    j += 1
                k += 1

            while i < len(left):
                arr[k] = left[i]
                i += 1
                k += 1

            while j < len(right):
                arr[k] = right[j]
                j += 1
                k += 1
            
        return arr

    def heapify(self, n, i):
        largest = i
        left = 2 * i + 1
        right = 2 * i + 2

        if left < n and self.array[left] > self.array[largest]:
            largest = left
        
        if right < n and self.array[right] > self.array[largest]:
            largest = right

        if largest != i:
            self.array[i], self.array[largest] = self.array[largest], self.array[i]

            self.heapify(n, largest)

    def heapSort(self):
        n = len(self.array)

        for i in range(n // 2 - 1, -1, -1):
            self.heapify(n, i)

        for i in range(n - 1, 0, -1):
            self.array[0], self.array[i] = self.array[i], self.array[0]

            self.heapify(i, 0)
        return self.array

        
    def run(self):
        self.enterArray()
        print(f"Przed sortowaniem: {self.array}")
        self.bubbleSort()
        print(f"bubbleSort: Po sortowaniu: {self.array}")
        self.selectionSort()
        print(f"selectionSort: Po sortowaniu: {self.array}")
        self.insertionSort()
        print(f"insertionSort: Po sortowaniu: {self.array}")
        self.mergeSort()
        print(f"mergeSort: Po sortowaniu: {self.array}")
        self.heapSort()
        print(f"heapSort: Po sortowaniu: {self.array}")

if __name__ == "__main__":
    app = Sorting()
    app.run()

File: sort/react/src/assets/scripts/sort.js
export function bubbleSort() {
    let userInput = document.getElementById("userInput").value;
    const array = userInput
        .trim()
        .split(/[\s,]+/)
        .map(numStr => Number(numStr))
        .filter(num => !isNaN(num));

    if (array.length === 0) {
        document.getElementById("result").innerHTML = "Wprowadź poprawne liczby!";
        return;
    }

    let n = array.length;
    for (let i = 0; i < n - 1; i++) {
        for (let j = 0; j < n - 1 - i; j++) {
            if (array[j] > array[j + 1]) {
                [array[j], array[j + 1]] = [array[j + 1], array[j]];
            }
        }
    }
    const result = array.join(", ");
    document.getElementById("result").innerHTML = result;
}

export function selectionSort() {
    let userInput = document.getElementById("userInput").value;
    const array = userInput
        .trim()
        .split(/[\s,]+/)
        .map(numStr => Number(numStr))
        .filter(num => !isNaN(num));

    if (array.length === 0) {
        document.getElementById("result").innerHTML = "Wprowadź poprawne liczby!";
        return;
    }
    let n = array.length;

    for (let i = 0; i < n - 1; i++) {
        let min = i;
        for (let j = i + 1; j < n; j++) {
            if (array[j] < array[min]) {
                min = j;
            }
        }
        if (min !== i) {
            [array[i], array[min]] = [array[min], array[i]];
        }
    }
    const result = array.join(", ");
    document.getElementById("result").innerHTML = result;

}

export function insertionSort() {
    let userInput = document.getElementById("userInput").value;
    const array = userInput
        .trim()
        .split(/[\s,]+/)
        .map(numStr => Number(numStr))
        .filter(num => !isNaN(num));

    if (array.length === 0) {
        document.getElementById("result").innerHTML = "Wprowadź poprawne liczby!";
        return;
    }

    let n = array.length;
    for (let i = 1; i < n; i++) {
        let key = array[i];
        let j = i - 1;
        while (j >= 0 && key < array[j]) {
            array[j + 1] = array[j];
            j -= 1;
        }
        array[j + 1] = key;
    }
    const result = array.join(", ");
    document.getElementById("result").innerHTML = result;
}

export function mergeSort(array) {
    let isInitial = false;

    if (!Array.isArray(array)) {
        isInitial = true;
        let userInput = document.getElementById("userInput").value;
        array = userInput
            .trim()
            .split(/[\s,]+/)
            .map(numStr => Number(numStr))
            .filter(num => !isNaN(num));
        if (array.length === 0) {
            document.getElementById("result").innerHTML = "Wprowadź poprawne liczby!";
            return;
        }
    }

    if (array.length > 1) {
        let mid = Math.floor(array.length / 2);
        let left = array.slice(0, mid);
        let right = array.slice(mid);

        mergeSort(left);
        mergeSort(right);
        let i = 0;
        let j = 0;
        let k = 0;
        while (i < left.length && j < right.length) {
            if (left[i] < right[j]) {
                array[k] = left[i];
                i += 1;
            } else {
                array[k] = right[j];
                j += 1;
            }
            k += 1;
        }

        while (i < left.length) {
            array[k] = left[i];
            i += 1;
            k += 1;
        }

        while (j < right.length) {
            array[k] = right[j];
            j += 1;
            k += 1;
        }
    }
    if (isInitial) {
        const result = array.join(", ");
        document.getElementById("result").innerHTML = result;
    }
}
export function heapify(array, n, i) {
    let largest = i;
    let left = 2 * i + 1;
    let right = 2 * i + 2;

    if (left < n && array[left] > array[largest]) {
        largest = left;
    }
    if (right < n && array[right] > array[largest]) {
        largest = right;
    }
    if (largest !== i) {
        [array[i], array[largest]] = [array[largest], array[i]];

        heapify(array, n, largest);
    }
}
export function heapSort() {
    let userInput = document.getElementById("userInput").value;
    const array = userInput
        .trim()
        .split(/[\s,]+/)
        .map(numStr => Number(numStr))
        .filter(num => !isNaN(num));

    if (array.length === 0) {
        document.getElementById("result").innerHTML = "Wprowadź poprawne liczby!";
        return;
    }
    let n = array.length;

    for (let i = Math.floor(n / 2) - 1; i >= 0; i--) {
        heapify(array, n, i);
    }


    for (let i = n - 1; i > 0; i--) {
        [array[0], array[i]] = [array[i], array[0]];
        heapify(array, i, 0);
    }

    const result = array.join(", ");
    document.getElementById("result").innerHTML = result;
}